Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1261,6 +1262,20 @@ void LocalFrame::frameWasDisconnectedFromOwner() const
     protectedDocument()->detachFromFrame();
 }
 
+void LocalFrame::didAccessWindowProxyPropertyViaOpener(WindowProxyProperty property)
+{
+    if (m_accessedWindowProxyPropertiesViaOpener.contains(property))
+        return;
+    m_accessedWindowProxyPropertiesViaOpener.add(property);
+
+    auto* parentWindow = opener() ? opener()->window() : nullptr;
+    if (!parentWindow)
+        return;
+
+    auto parentOrigin = SecurityOriginData::fromURL(parentWindow->location().url());
```

Comment:
I always get worried about opaque origins in when doing these kind of things. Just mentioning it so that you can take it into consideration.

Reply:
Good point. I'll skip null and opaque origins here